openstack/neutron/templates/statefulset-network-agent-apod.yaml

```diff
@@ -141,10 +165,32 @@ spec:
               value: "ignore:Unverified HTTPS request"
           readinessProbe:
             exec:
-              command: ["neutron-linuxbridge-readiness", "-config-file", "/etc/neutron/neutron.conf"]
-            initialDelaySeconds: 30
-            periodSeconds: 60
+              command:
+              - python
+              - -c
+              - |
+                import os
+                import requests
+
+                PROMETHEUS = "http://prometheus-openstack.prometheus-openstack:9090"
+                QUERY = """
+                  sum(openstack_subnets_with_dhcp_ip_allocation{network_host="%s"}) by (bridge, network_host)
+                    unless on(bridge, network_host)
+                  label_replace(neutron_agent_checks_linux_bridge_present, "network_host", "$1", "hostname", "(.*)")
+                """
+                response =requests.get(PROMETHEUS + "/api/v1/query", params={"query": QUERY % os.uname()[1]}) 
+                if  response.status_code != 200:
+                    print(f"Error while running prometheus query: {response.json()}")
+                    exit(-1)
```

Comment:
there is no negative program exit code

Also, maybe you don't want to fail  here:
If prometheus is down and neutron is being rolled out, kubernetes would wait indefinitely to bring up new neutron agent pods since they would stay unready. Jusy maybe taking it into consideration.

Reply:
ACK on the exit code even though it just get 2**16 iirc.
To the other topic: if we'd succeed here, kubernetes would just roll all agent pods quickly in sequence thus creating an outage. So I was thinking that being stuck is better than rolling in quickly and creating an outage.